### PR TITLE
changed weights to 1/sigma^2, fixed chi^2, and clarified the documentation

### PR DESCRIPTION
**Pull request by jsdillon**

### CODE REVIEW

File: linsolve/linsolve.py [hunk 1]
```diff
@@ -305,13 +307,16 @@ def get_A_sparse(self):
...
-        '''Return y = data * wgt as a 2D vector, regardless of original data/wgt shape.'''
+        '''Return y = data * wgt**.5 as a 2D vector, regardless of original data/wgt shape.'''
         d = np.array([self.data[k] for k in self.keys])
         if len(self.wgts) > 0:
             w = np.array([self.wgts[k] for k in self.keys])
             w.shape += (1,) * (d.ndim-w.ndim)
             d.shape += (1,) * (w.ndim-d.ndim)
-            d = d*w
+            d = d*(w**.5) 
+            # this is w**.5 because A alredy has a factor of w**.5 in it, so
```
Comment (AaronParsons) [inline linsolve/linsolve.py]: already

File: linsolve/linsolve.py [hunk 2]
```diff
@@ -446,7 +451,8 @@ def __init__(self, data, wgts={}, sparse=False, **kwargs):
                 Variables with trailing underscores '_' are interpreted as complex conjugates (e.g. x*y_ 
                 parses as x * y.conj()).
             wgts: Dictionary that maps equation strings from data to real weights to apply to each 
-                equation. Any equation not in wgts gets a weight of 1.0. Defaults to {} (i.e. all 1.0s).
+                equation. Weights are treated as 1/sigma^2. Any equation not in wgts gets a weight of 1.0.
```
Comment (AaronParsons) [inline linsolve/linsolve.py]: Seems to me if you provide wgts, it should be an error not to have an equation assigned a weight.

Comment (jsdillon) [reply linsolve/linsolve.py]: Turns out that's actually what the code was doing and the documentation was just out of date. I'll fix it.

Comment (jsdillon) [reply linsolve/linsolve.py]: (see the verify_weights function)

**Comment by jsdillon** [comment]
@AaronParsons I believe I've addressed your request for changes.

File: linsolve/linsolve.py (hunk 1 above)
Comment (jsdillon) [reply linsolve/linsolve.py]: Fixed.

**Comment by jsdillon** [comment]
@AaronParsons approved this in person

**Pull request merged**